dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/datafile/FilePermissionsServiceIT.java

```diff
@@ -0,0 +1,233 @@
+        DataFile datafile = dataFileService.find(53L);
+        
+        // when
+        Executable assignRoleOperation = () -> filePermissionsService.assignFileDownloadRole(
+                Lists.newArrayList(user),
+                Lists.newArrayList(datafile));
+        
+        // then
+        assertThrows(PermissionException.class, assignRoleOperation);
+    }
+    
+    
+    @Test
+    @Transactional(TransactionMode.ROLLBACK)
+    public void revokeFileDownloadRole() {
+        // given
+        DataFile datafile1 = dataFileService.find(53L);
+        DataFile datafile2 = dataFileService.find(55L);
+        AuthenticatedUser user = authenticationService.getAuthenticatedUser("filedownloader");
+        
+        // when
+        List<RoleAssignment> removedRoleAssignments = filePermissionsService.revokeFileDownloadRole(
+                Lists.newArrayList(user),
+                Lists.newArrayList(datafile1, datafile2));
+        
+        // then
+        assertEquals(2, removedRoleAssignments.size());
+        
+        assertContainsRoleAssignment(removedRoleAssignments, "@filedownloader", 53L);
+        assertContainsRoleAssignment(removedRoleAssignments, "@filedownloader", 55L);
+        
+        assertThat(roleAssigneeService.getAssignmentsFor("@filedownloader"), is(empty()));
+    }
+    
+    
+    @Test
+    @Transactional(TransactionMode.ROLLBACK)
+    public void revokeFileDownloadRole__MISSING_MANAGE_DATASET_PERMISSIONS() {
+        // given
+        dataverseSession.setUser(GuestUser.get());
+        DataFile datafile = dataFileService.find(53L);
+        AuthenticatedUser user = authenticationService.getAuthenticatedUser("filedownloader");
+        
+        // when
+        Executable revokeRoleOperation = () -> filePermissionsService.revokeFileDownloadRole(Lists.newArrayList(user), Lists.newArrayList(datafile));
+        
+        // then
+        assertThrows(PermissionException.class, revokeRoleOperation);
+    }
+    
+    
+    @Test
+    public void rejectRequestAccessToFiles() {
+        // given
+        AuthenticatedUser user = authenticationService.getAuthenticatedUser("superuser");
+        
+        DataFile datafile1 = dataFileService.find(53L);
+        datafile1.getFileAccessRequesters().add(user);
+        datafile1 = dataFileService.save(datafile1);
+        
+        DataFile datafile2 = dataFileService.find(55L);
+        datafile2.getFileAccessRequesters().add(user);
+        datafile2 = dataFileService.save(datafile2);
+        
+        // when
+        filePermissionsService.rejectRequestAccessToFiles(user, Lists.newArrayList(datafile1, datafile2));
+        
+        // then
+        assertThat(dataFileService.find(53L).getFileAccessRequesters(), is(empty()));
+        assertThat(dataFileService.find(55L).getFileAccessRequesters(), is(empty()));
+        
+        EmailModel userEmail = waitForEmailSentTo(smtpServer, "superuser@mailinator.com");
```

Comment:
Can we use class.method call for FakeSmtpServerUtil.waitForEmailSentTo() for clarity?